lua/cfc_time/server/time/sv_mysql.lua

```diff
@@ -0,0 +1,217 @@
+    CFCTime.Logger:info( "Constructing prepared statements..." )
+
+    local realm = self.realm
+
+    local newUser = "INSERT IGNORE INTO users (steam_id) VALUES(?)"
+
+    local newSession = string.format( [[
+        INSERT INTO sessions (user_id, start, realm) VALUES(?, ?, %s)
+    ]], realm )
+
+    local totalTime = string.format( [[
+        SELECT SUM(duration)
+        FROM sessions
+        WHERE user_id = ?
+        AND realm = %s
+    ]], realm )
+
+    self:AddPreparedStatement( "newUser", newUser )
+    self:AddPreparedStatement( "newSession", newSession )
+    self:AddPreparedStatement( "totalTime", totalTime )
+end
+
+function CFCTime.SQL:Prepare( statementName, onSuccess, ... )
+    local query = self.preparedStatements[statementName]
+    query:clearParameters()
+
+    for k, v in pairs( { ... } ) do
+        if isnumber( v ) then
+            query:setNumber( k, v )
+        elseif isstring( v ) then
+            query:setString( k, v )
+        elseif isbool( v ) then
+            query:setBoolean( k, v )
+        else
+            error( "Wrong data type passed to Prepare statement!: " .. v )
+        end
+    end
+
+    query.onSuccess = onSuccess or noop
+
+    return query
+end
+
+function CFCTime.SQL.database:onConnected()
+    CFCTime.Logger:info( "DB successfully connected! Beginning init..." )
+
+    local transaction = CFCTime.SQL:InitTransaction()
+
+    transaction:addQuery( CFCTime.SQL:CreateUsersQuery() )
+    transaction:addQuery( CFCTime.SQL:CreateSessionsQuery() )
+    transaction:addQuery( CFCTime.SQL:EndTimeCleanupQuery() )
+
+    transaction.onSuccess = function()
+        CFCTime.SQL:PrepareStatements()
+    end
+
+    transaction:start()
+end
+
+function CFCTime.SQL.database:onConnectionFailed( _, err )
+    CFCTime.Logger:error( "Failed to connect to database!" )
+    CFCTime.Logger:fatal( err )
+end
+
+hook.Add( "PostGamemodeLoaded", "CFC_Time_DBInit", function()
+    CFCTime.Logger:log( "Gamemoded loaded, beginning database init..." )
+    CFCTime.SQL.database:connect()
+end )
+
+function CFCTime.SQL:BuildSessionUpdate( data, id )
+    local updateSection = "UPDATE sessions "
+    local setSection = "SET "
+    local whereSection = "WHERE id = " .. id
+    local whereSection = string.format(
+        "WHERE id = %s AND realm = %s",
+        id, self.realm
+    )
+
+    local count = table.Count( data )
+    local idx = 1
+    for k, v in pairs( data ) do
+        local newSet = k .. " = " .. v
+        
+        if idx ~= count then
+            -- Add a comma if it isn't the last one
+            newSet = newSet .. ","
+        else
+            -- Add a space if it's the last one
+            newSet = newSet .. " "
+        end
+        
+        setSection = setSection .. newSet
+        idx = idx + 1
+    end
+
+    local query = updateSection .. setSection .. whereSection
+
+    return query
+end
+
+--[ API Begins Here ]--
+
+function CFCTime.SQL:UpdateBatch( batchData )
```

Comment:
could call this CFCTime.Storage instead of CFCTime.SQL because non sql storage options could be  created